projects/lib/src/providers/google-login-provider.ts

```diff
@@ -44,25 +44,24 @@ export class GoogleLoginProvider extends BaseLoginProvider {
     });
   }
 
-  getLoginStatus(): Promise<SocialUser> {
+  getLoginStatus(refreshToken = false): Promise<SocialUser> {
     return new Promise((resolve, reject) => {
       if (this.auth2.isSignedIn.get()) {
         let user: SocialUser = new SocialUser();
 
-        let profile = this.auth2.currentUser.get().getBasicProfile();
-        let token = this.auth2.currentUser.get().getAuthResponse(true)
-          .access_token;
-        let backendToken = this.auth2.currentUser.get().getAuthResponse(true)
-          .id_token;
+        const profile = this.auth2.currentUser.get().getBasicProfile();
+        const authResponse = refreshToken ?
+          this.auth2.currentUser.get().reloadAuthResponse(true) :
+          this.auth2.currentUser.get().getAuthResponse(true);
```

Comment:
Hey there, the reloadAuthResponse returns a promise other than a general response. I've done a little bit tweaking locally and it works fine.

      ```
  let user: GoogleUser = new GoogleUser();
        const profile = this.auth2.currentUser.get().getBasicProfile();
        user.id = profile.getId();
        user.name = profile.getName();
        user.email = profile.getEmail();
        user.photoUrl = profile.getImageUrl();
        user.firstName = profile.getGivenName();
        user.lastName = profile.getFamilyName();
        user.response = profile;

        if(refreshToken)
        {
            this.auth2.currentUser.get()
                .reloadAuthResponse(true)
                .then(response=>{
                    user.authToken = response.access_token;
                    user.idToken = response.id_token;
                    resolve(user);
                })
        }
        else
        {
            const authResponse = this.auth2.currentUser.get().getAuthResponse(true);
            user.authToken = authResponse.access_token;
            user.idToken = authResponse.id_token;
            resolve(user);
        }
```

Reply:
@jaibatrik @carbon-wpa Btw, thank you guys for making this happen! Since token is valid only for an hour, this feature is really helpful to not interrupt operations or upset people. We can put a setting in our apps for the valid login duration and build some logic around it to refresh the token when needed.

Reply:
I have applied small corrections to the code and waiting for the feedback 👍

Reply:
This looks good to me. Although, I won't place the check of whether a provider supports refreshing token at `SocialAuthService` level. I'll probably leave that to individual providers. I'll make that refactor later.